tests/catch/unit/memory/hipMemCoherencyTst.cc

```diff
@@ -59,34 +51,30 @@ __global__  void SquareKrnl(int *ptr) {
 static bool YES_COHERENT = false;
 
 // The function tests the coherency of allocated memory
-static void TstCoherency(int *Ptr, bool HmmMem) {
-  int *Dptr = nullptr, peak_clk;
-  hipStream_t strm;
-  HIP_CHECK(hipStreamCreate(&strm));
+// If this test hangs, means there is issue in coherency
+static void TstCoherency(int* ptr, bool hmmMem) {
+  int* dptr = nullptr;
+  hipStream_t stream{};
+  HIP_CHECK(hipStreamCreate(&stream));
+
   // storing value 1 in the memory created above
-  *Ptr = 1;
-  // Getting gpu frequency
-  HIP_CHECK(hipDeviceGetAttribute(&peak_clk, hipDeviceAttributeClockRate, 0));
-  if (!HmmMem) {
-    HIP_CHECK(hipHostGetDevicePointer(reinterpret_cast<void **>(&Dptr), Ptr,
-                                      0));
-    CoherentTst<<<1, 1, 0, strm>>>(Dptr, peak_clk);
+  *ptr = 1;
+
+  if (!hmmMem) {
+    HIP_CHECK(hipHostGetDevicePointer(reinterpret_cast<void**>(&dptr), ptr, 0));
+    CoherentTst<<<1, 1, 0, stream>>>(dptr);
   } else {
-    CoherentTst<<<1, 1, 0, strm>>>(Ptr, peak_clk);
-  }
-  // looping until the value is 2 for 3 seconds
-  std::chrono::steady_clock::time_point start =
-               std::chrono::steady_clock::now();
-  while (std::chrono::duration_cast<std::chrono::seconds>(
-         std::chrono::steady_clock::now() - start).count() < 3) {
-    if (*Ptr == 2) {
-      *Ptr += 1;
-      break;
-    }
+    CoherentTst<<<1, 1, 0, stream>>>(ptr);
   }
-  HIP_CHECK(hipStreamSynchronize(strm));
-  HIP_CHECK(hipStreamDestroy(strm));
-  if (*Ptr == 4) {
+
+  while (*ptr == 2) {
```

Comment:
keep a timeout here, we don't want tests being stuck.

Reply:
Done